pkg/virt-controller/watch/application.go

```diff
@@ -232,14 +233,14 @@ func (vca *VirtControllerApp) Run() {
 			RenewDeadline: vca.LeaderElection.RenewDeadline.Duration,
 			RetryPeriod:   vca.LeaderElection.RetryPeriod.Duration,
 			Callbacks: leaderelection.LeaderCallbacks{
-				OnStartedLeading: func(stopCh <-chan struct{}) {
-					vca.informerFactory.Start(stop)
-					go vca.nodeController.Run(controllerThreads, stop)
-					go vca.vmiController.Run(controllerThreads, stop)
-					go vca.rsController.Run(controllerThreads, stop)
-					go vca.vmController.Run(controllerThreads, stop)
-					go vca.migrationController.Run(controllerThreads, stop)
-					cache.WaitForCacheSync(stopCh, vca.persistentVolumeClaimInformer.HasSynced)
+				OnStartedLeading: func(ctx context.Context) {
+					vca.informerFactory.Start(ctx.Done())
+					go vca.nodeController.Run(ctx, controllerThreads)
```

Comment:
I understand the leader election callback requires a context now rather than just a stop channel. What's unclear to me is the value of passing the the context into these Controllers now rather than just the stop channel like we did previously?

If there's not a good reason to be passing the context around to all our controllers, I'd prefer not to make that change here since we're primarily interested in vendoring k8s 1.12.4. We could keep things the way they were by just passing in ctx.Done() right?

Reply:
To be honest there isn't a good reason to pass the context around. I just thought having the context available in the controllers might be a good idea to build on in the future.
But you're right, the goal of this PR is venturing k8s 1.12.4. So we may keep the interfaces as they are.